modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -315,15 +315,13 @@ public static IgniteState state(@Nullable String name) {
      *      execution. If {@code false}, then jobs currently running will not be
      *      canceled. In either case, grid node will wait for completion of all
      *      jobs running on it before stopping.
-     * @param waitForBackups If {@code true} then node will wait until all of its data is backed up before shutting
-     *      down. Please note that it will completely prevent last node in cluster from shutting down if any caches
-     *      exist that have backups configured. If {@code null} then
-     *      {@link IgniteSystemProperties#IGNITE_WAIT_FOR_BACKUPS_ON_SHUTDOWN} value is used instead.
+     * @param shutdown If this parameter sets explicitly this policy will use for stopping node.
+     *       If this parameter is {@code null} common cluster policy will use.
      * @return {@code true} if default grid instance was indeed stopped,
      *      {@code false} otherwise (if it was not started).
      */
-    public static boolean stop(boolean cancel, @Nullable Boolean waitForBackups) {
-        return stop(null, cancel, waitForBackups, false);
+    public static boolean stop(boolean cancel, @Nullable ShutdownPolicy shutdown) {
```

Comment:
Why not to encapsulate canceling jobs into ShutdownPolicy?
 
Does it guarantee that CheckCpHistClosureJob will run in case of cancel true?

Reply:
This was discusses in developer list. The flag can to be in shutdown policies, but in the future.
Now, the policies are responding for possible data loosing.

Yes, the job executing before node stopping, so that cancel flag does not influence this job.